File: Gauss/src/Datatypes/MatrixDouble.java
package Datatypes;

public class MatrixDouble implements MatrixDataType<MatrixDouble,Double> {

    private double value;

    public MatrixDouble(int v) {
        setValue(((double)v)/0x10000); //0x10000 = 2^16
    }

    public MatrixDouble(double v) { this.value=v; }

    @Override
    public double evaluate() {
        return this.value;
    }

    @Override
    public Double getValue() {
        return this.value;
    }

    @Override
    public void setValue(Double v) {
        this.value = v;
    }

    @Override
    public String toString() { return ""+value;}

    @Override
    public MatrixDouble add(MatrixDouble number) {
        return new MatrixDouble(this.getValue() + number.getValue());
    }

    @Override
    public MatrixDouble subtract(MatrixDouble number) {
        return new MatrixDouble(this.getValue() - number.getValue());
    }

    @Override
    public MatrixDouble multiply(MatrixDouble number) {
        return new MatrixDouble(this.getValue() * number.getValue());
    }

    @Override
    public MatrixDouble divide(MatrixDouble number) {
        return new MatrixDouble(this.getValue() / number.getValue());
    }
}

File: Approximation/Approximation/Matrix/Datatypes/MatrixDouble.cs
namespace Approximation.Matrix.Datatypes
{
    public class MatrixDouble : IMatrixDataType<double> {
        public static readonly MatrixDouble ONE = new MatrixDouble(1.0);
        public static readonly MatrixDouble MINUSONE = new MatrixDouble(-1.0);
        public static readonly MatrixDouble ZERO = new MatrixDouble(0.0);

        private double Value { get; set; }

        public MatrixDouble(double v) {
            Value = v;
        }
   
        public double Evaluate() {
            return Value;
        }

        public double GetValue() {
            return Value;
        }

        public void SetValue(double v) {
            Value = v;
        }

        public override string ToString() {
            return Value.ToString();
        }

        public IMatrixDataType<double> Add(IMatrixDataType<double> number) {
            return new MatrixDouble(GetValue() + number.GetValue());
        }

        public IMatrixDataType<double> Subtract(IMatrixDataType<double> number) {
            return new MatrixDouble(GetValue() - number.GetValue());
        }

        public IMatrixDataType<double> Multiply(IMatrixDataType<double> number) {
            return new MatrixDouble(GetValue() * number.GetValue());
        }

        public IMatrixDataType<double> Divide(IMatrixDataType<double> number) {
            return new MatrixDouble(GetValue() / number.GetValue());
        }

        public IMatrixDataType<double> GetInverse() {
            return ONE.Divide(this);
        }

        public int CompareTo(IMatrixDataType<double> number) {
            return GetValue().CompareTo(number.GetValue());
        }

        public IMatrixDataType<double> Abs() {
            if (CompareTo(ZERO) > 0)
                return this;
            else
                return Multiply(MINUSONE);
        }
    }
}